src/main/java/net/bootsfaces/component/slider2/Slider2Renderer.java

```diff
@@ -93,35 +80,67 @@ public void encodeBegin(FacesContext context, UIComponent component) throws IOEx
 		encodeHTML(slider, context, rw);
 	}
 
+	/**
+	 * Returns converted value of the provided float value (to the type that is used on the component's value
+	 * attribute).
+	 *
+	 * @param component Component to convert value for.
+	 * @param value     Float value to convert.
+	 *
+	 * @return Converted value of the provided float value (to the type that is used on the component's value
+	 *         attribute).
+	 */
+	private Number convert(UIComponent component, Float value) {
+		return convert((Class<? extends Number>) getValueType(component), value);
+	}
+
+	/**
+	 * Returns converted value of the provided float value (to the provided type).
+	 *
+	 * @param type  Type to convert float to. Currently only {@link Float} and {@link Integer} are supported.
+	 * @param value Float value to convert.
+	 *
+	 * @return Converted value of the provided float value (to the provided type).
+	 *
+	 * @throws IllegalArgumentException If an unsupported type is provided.
+	 */
+	private Number convert(Class<? extends Number> type, Float value) {
+		if (int.class.isAssignableFrom(type) || Integer.class.isAssignableFrom(type)) {
+			return value.intValue();
+		}
+		if (float.class.isAssignableFrom(type) || Float.class.isAssignableFrom(type)) {
+			return value;
+		}
+		throw new IllegalArgumentException("Use integer or float");
+	}
+
 	private void encodeHTML(Slider2 slider, FacesContext context, ResponseWriter rw) throws IOException {
 		String clientId = slider.getClientId(context);
 
 		String mode = slider.getMode();
 		String label = slider.getLabel();
+		Class<? extends Number> type = (Class<? extends Number>) getValueType(slider);
+		if (type == null) {
+			type = Integer.class;
+		}
 
-		int min = slider.getMin();
-		int max = slider.getMax();
+		float min = ((Number) slider.getMin()).floatValue();
+		float max = ((Number) slider.getMax()).floatValue();
 		Object v = slider.getSubmittedValue();
 		if (v == null) {
 			v = slider.getValue();
 		}
 		if (v == null) {
-			v = slider.getValue();
-			slider.setValue(v);
-		}
-		if (v == null) {
-			v = max / 2;
-			slider.setValue(v);
+			v = min + (max - min) / 2;
```

Comment:
That's even a small bug fix, isn't it? :)

Reply:
Yes it was.. see https://github.com/TheCoder4eu/BootsFaces-OSP/issues/976#issuecomment-407347738